core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -211,7 +211,7 @@ abstract class RDD[T: ClassTag](
   // Our dependencies and partitions will be gotten by calling subclass's methods below, and will
   // be overwritten when we're checkpointed
   private var dependencies_ : Seq[Dependency[_]] = null
-  @transient private var partitions_ : Array[Partition] = null
+  @transient @volatile private var partitions_ : Array[Partition] = null
```

Comment:
`partitions_ ` would be write/read by multiple threads, just put it here for visibility.